NARFO_BE/NARFO_BE/Controllers/MembersController.cs

```diff
@@ -2,121 +2,133 @@
 
         public MembersController(narfoContext context)
         {
             _context = context;
             if (!_context.Members.Any())
             {
-                Members newMember = new Members();
-                newMember.LastName = "admin";
+                _Member newMember = new _Member();
+                newMember.SURNAME = "admin";
                 newMember.Username = "admin";
-                newMember.FirstName = "admin";
-                newMember.Password = "admin";
+                newMember.Firstname = "admin";
+                newMember.Email = "admin@admin.com";
+               
                 _context.Members.AddAsync(newMember);
                 _context.SaveChanges();
             }
         }
-        
-        public void ListofMembers(List<Members> members)
-        {
-            this.members = members;
-        }
-        
 
-        // GET: api/Members
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<Members>>> GetMembers()
-        {
-            return await _context.Members.ToListAsync();
-        }
-        public IEnumerable<Members> GetAllMembers()
+        public void ListofMembers(List<_Member> members)
         {
-            return members;
+            this.members = members;
         }
 
-        // GET: api/Members/5
-        [HttpGet("{id}")]
-        public async Task<ActionResult<Members>> GetMembers(int id)
+        //
+        // POST: /Account/Login
+        [HttpPost("get/login")]
+        public async Task<ActionResult <_Member>> Login([FromBody] _Member model)
         {
-            var members = await _context.Members.FindAsync(id);
+            
+            _Member user = await _context.Members.FirstOrDefaultAsync(member=>member.Email == model.Email && member.Password == encryption.ComputeHash(model.Password));
 
-            if (members == null)
+            if (user == null)
             {
-                return NotFound();
+                return BadRequest(new { status = "Failed", message = "Invalid login" });
             }
+            else {
 
-            return members;
+                return Ok(new { status = "Success", member = model });
+            }
+            
+            
         }
 
-        // PUT: api/Members/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutMembers(int id, Members members)
+        private ActionResult<_Member> Json(object p)
         {
-            if (id != members.Id)
-            {
-                return BadRequest();
-            }
+            throw new NotImplementedException();
+        }
 
-            _context.Entry(members).State = EntityState.Modified;
 
-            try
+        // GET: Member/Email
+        [HttpGet("get/all/user")]
+        public  async Task<ActionResult<MemberPrototype>> GetMembersEmail()
+        {
+           
+           List<MemberPrototype> endpoint = new List<MemberPrototype>();
+            foreach (_Member d in  await _context.Members.ToArrayAsync())
```

Comment:
> This is just a comment

You want

```
var endpoint = await _context.Members.Where(d => d.Username !=  null && d.Email != null).Select(d => new 
 MemberPrototype(d.Username,d.Email)).ToArrayAsync();
return enpoint.Count > 0 ? Ok(new { status = "success", members=endpoint }); : BadRequest(new { status = "failed", error = "Failed to connect" });
```

?

> just saying 😃

Reply:
thanks  for the quicker way of doing things . but MemberPrototype and _Member are not the same types... two objects envolved "_Member" and "MemberPrototype"  unless if your query works with different objects????? :

Reply:
Thats why we have `.Select(d => new 
 MemberPrototype(d.Username,d.Email)`

Reply:
thank you for your view ( : we'll do a followup